logstash-core/src/main/java/org/logstash/instrument/witness/QueueWitness.java

```diff
@@ -36,10 +43,49 @@ public void type(String type) {
         this.type.set(type);
     }
 
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        new Serializer().innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    public static class Serializer extends StdSerializer<QueueWitness> {
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(QueueWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<QueueWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(QueueWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
+
+        void innerSerialize(QueueWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeObjectFieldStart(witness.KEY);
```

Comment:
@jakelandis won't weird things happen here, if our custom serializer from `MetricSerializer.Get` at times just does nothing but you still write an object field start?